source/common/upstream/upstream_impl.cc

```diff
@@ -651,6 +651,14 @@ ClusterInfoImpl::extensionProtocolOptions(const std::string& name) const {
   return nullptr;
 }
 
+const ConnectionRequestPolicy& ClusterInfoImpl::connectionPolicy() const {
+  if (!connection_policy_) {
```

Comment:
I doubt this is thread safe? But it's clear to me yet where this is used and whether it needs to live here.

Reply:
Not in its current form. The idea is that the Policy implementation would be `injected` but I am not sure how that would be implemented. I might need help with that.